File: cpp/n189_RotateArray.cpp
// https://leetcode-cn.com/problems/rotate-array/

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution{
public:
    void rotate(vector<int>& nums, int k){
        int length = nums.size();
        k = k % length;
        reverse(&nums[0], &nums[length]);
        reverse(&nums[0], &nums[k]);
        reverse(&nums[k], &nums[length]);
    }
};

int main(){
    return 0;
}

File: python/n189_RotateArray.py
# https://leetcode-cn.com/problems/rotate-array/

class Solution():
    def rotate(self, nums, k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k = k % len(nums)
        if k == 0:
            pass
        else:
            nums[:] = nums[-k:] + nums[:-k]
            

if __name__ == "__main__":
    pass